api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -167,6 +167,10 @@ const (
 	// of the cluster will be removed, including image registry storage, ingress dns records, load balancers, and persistent storage.
 	CleanupCloudResourcesAnnotation = "hypershift.openshift.io/cleanup-cloud-resources"
 
+	// CleanupOrcImageResourcesAnnotation is an annotation that indicates whether an OpenStack Image should be
+	// removed when deleting the corresponding ORC Image. If set to "false", the OpenStack Image will not be removed.
+	CleanupOrcImageResourcesAnnotation = "hypershift.openshift.io/cleanup-orc-image-resource"
```

Comment:
I would rename it to `CleanupORCImageResourcesAnnotation`.

Constants in Go typically follow [CamelCase](https://github.com/zecke/Kubernetes/blob/master/docs/devel/api-conventions.md#constants) and avoid underscores. Also, acronyms (like ORC) are usually all uppercase to maintain readability.